x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/ApiKeyService.java

```diff
@@ -1744,23 +1733,34 @@ AtomicLong getLastEvictionCheckedAt() {
      */
     public static String getCreatorRealmName(final Authentication authentication) {
         if (authentication.isApiKey()) {
-            return (String) authentication.getMetadata().get(AuthenticationField.API_KEY_CREATOR_REALM_NAME);
+            return (String) authentication.getAuthenticatingSubject().getMetadata().get(AuthenticationField.API_KEY_CREATOR_REALM_NAME);
```

Comment:
I think we should consistently use `getEffectiveSubject()` here as well. In practice, both `getEffectiveSubject` and `getAuthenticatingSubject` return the same object because API key cannot be the target of run-as. But using `getAuthenticatingSubject` is assuming that knowledge here which isn't great and can in theory become incorrect if we start supporting run-as an API Key. In general, code at service layer should mostly not have to worry about authenticatingSubject unless there is a strong reason for it.

I am aware that using `getAuthenticatingSubject` is more faithful to what the existing code does. But in this case, I think the existing code isn't quite right either.

Reply:
I'm good with using `getEffectiveSubject` here. I went with `getAuthenticatingSubject` because it was suggested by the deprecation Javadoc for [`getMetadata`](https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/Authentication.java#L226).